src/components/Annotation.tsx

```diff
@@ -0,0 +1,78 @@
+import {
+  useContext,
+  useEffect,
+  useState,
+  useRef,
+} from 'react';
+import { createPortal } from 'react-dom';
+import MapContext from '../context/MapContext';
+import AnnotationProps from './AnnotationProps';
+
+export default function Annotation({
+  latitude,
+  longitude,
+
+  title = '',
+  subtitle = '',
+  accessibilityLabel = null,
+
+  selected = undefined,
+  onSelect = undefined,
+  onDeselect = undefined,
+  children,
+}: AnnotationProps) {
+  const [annotation, setAnnotation] = useState<mapkit.Annotation | null>(null);
+  const contentRef = useRef<HTMLDivElement>(document.createElement('div'));
+  const map = useContext(MapContext);
+
+  // Coordinates
+  useEffect(() => {
+    if (map === null) return undefined;
+    if (!contentRef.current) return undefined;
+
+    const a = new mapkit.Annotation(
+      new mapkit.Coordinate(latitude, longitude),
+      () => contentRef.current,
+    );
+    map.addAnnotation(a);
+    setAnnotation(a);
+
+    return () => {
+      map.removeAnnotation(a);
+    };
+  }, [map, contentRef, latitude, longitude]);
+
+  // Simple values properties
+  const properties = {
+    title,
+    subtitle,
+    accessibilityLabel,
+
+    selected,
+  };
+  Object.entries(properties).forEach(([propertyName, prop]) => {
+    useEffect(() => {
+      if (!annotation) return;
+      // @ts-ignore
+      annotation[propertyName] = prop;
+    }, [annotation, prop]);
+  });
+
+  // Events
+  const events = [
+    { name: 'select', handler: onSelect },
+    { name: 'deselect', handler: onDeselect },
+  ] as const;
+  events.forEach(({ name, handler }) => {
+    useEffect(() => {
+      if (!annotation || !handler) return undefined;
+
+      const handlerWithoutParameters = () => handler();
+
+      annotation.addEventListener(name, handlerWithoutParameters);
+      return () => annotation.removeEventListener(name, handlerWithoutParameters);
+    }, [annotation, handler]);
+  });
+
+  return contentRef.current && createPortal(children, contentRef.current);
```

Comment:
Same thing as above, the check here isn’t necessary.
```suggestion
  return createPortal(children, contentRef.current);
```